protocol/consistencychecks.go

```diff
@@ -278,30 +329,33 @@ func (cc *ConsistencyChecks) verifySignedPromise(requestType int,
 	case RegistrationType:
 		if responseCode == ErrorNameExisted {
 			// ignore the key value verification since we may have no information about that existed binding.
-			if verifResult := cc.verifyReturnedPromise(tb, uname, nil, str, ap); verifResult != nil {
+			if verifResult := cc.verifyTB(tb, uname, nil, str, ap); verifResult != nil {
```

Comment:
I think it's fine to ignore the key in the TB, unless the TB is for the client's user. Then I think, the client should identify whether the TB is for its user, and ensure that the key in the TB still matches the expected key.